### PR TITLE
Docs/ipfs hash tests & fixer script

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

- Auto-detect ipfs hashes in our docs and fix them
- Test for ipfs hashes in the docs

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/echo_demo.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ This demo assumes you have followed the setup guide.
 The fastest way to have your first AEA is to fetch one that already exists!
 
 ``` bash
-aea fetch open_aea/my_first_aea:0.1.0 --local
```
Comment (DavidMinarsch) [inline docs/echo_demo.md:10]: Why are you removing the version?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Remove the svn command too please.

Also don't forget about `aea add`

File: docs/agent-vs-aea.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ In this guide we show some of the differences in terms of code.
 
 The <a href="../build-aea-programmatically">Build an AEA programmatically</a> guide shows how to programmatically build an AEA. We can build an agent of the <a href="../api/agent#agent-objects">`Agent`</a> class programmatically as well.
 
-First, import the python and application specific libraries. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
+First, import the python and application specific libraries. (Get the packages directory from the AEA repository `aea fetch PUBLIC_ID_OR_HASH`.)
```
Comment (dvilelaf) [inline docs/agent-vs-aea.md:9]: These svn commands were not specific to a package but a recommendation on how to get packages, so I just set this to a general command that i saw in other places in the docs.

Comment (DavidMinarsch) [inline docs/agent-vs-aea.md:9]:
```suggestion
First, import the python and application specific libraries. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
```
It makes no sense here now. If you read the sentence it has no place here as such. The whole guide doesn't make any sense anymore. It needs to be properly fixed.

File: docs/decision-maker-transaction.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 This guide can be considered as a part 2 of the <a href="../standalone-transaction/">the stand-alone transaction demo</a>. The main difference is that now we are going to use the decision-maker to sign the transaction.
 
-First, import the libraries and the set the constant values. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
+First, import the libraries and the set the constant values. (Get the packages directory from the AEA repository `aea fetch PUBLIC_ID_OR_HASH`.)
```
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md:3]: also doesn't make sense. You can see below it only uses one protocol. The `aea add ..` command could be used. But then it's in an agent project. Need to find another way.

File: docs/echo_demo.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ This demo assumes you have followed the setup guide.
...
+aea fetch fetchai/my_first_aea:0.27.0:QmcmunuEAvaKfLCJsgeqF8M5dKARHixu4otUFVGbEenaCm --remote
```
Comment (DavidMinarsch) [inline docs/echo_demo.md:10]:
```suggestion
aea fetch open_aea/my_first_aea:0.1.0:QmbyFaoHWikC8dHcpdTVboxVnkXX2vMgch4UUXmwCfJhRu --remote
```

File: docs/multiplexer-standalone.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 The `Multiplexer` can be used stand-alone. This way a developer can utilise the protocols and connections independent of the `Agent` or `AEA` classes.
 
-First, import the Python and application specific libraries and set the static variables. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
+First, import the Python and application specific libraries and set the static variables. (Get the packages directory from the AEA repository `aea fetch PUBLIC_ID_OR_HASH`.)
```
Comment (DavidMinarsch) [inline docs/multiplexer-standalone.md:3]: Same problem as above

File: docs/quickstart.md [hunk 6]
```diff
@@ -150,20 +150,20 @@ aea init --local
 AEAs are composed of components. AEAs and AEA components can be developed by anyone and pushed to an <a href="https://ipfs.io/" target="_blank">IPFS registry</a> for others to use.
 
 
-To load Valory packages please use <a href="https://subversion.apache.org/packages.html" target="_blank">SVN</a> to checkout the specific folders;
+To load Valory packages please use <a href="https://subversion.apache.org/packages.html" target="_blank">aea fetch</a> to checkout the specific packages;
```
Comment (DavidMinarsch) [inline docs/quickstart.md:153]:
```suggestion
```
@dvilelaf please read the whole para and then fix.

File: docs/quickstart.md [hunk 7]
```diff
@@ -150,20 +150,20 @@ aea init --local
...
 ```bash
```
Comment (DavidMinarsch) [inline docs/quickstart.md:154]:
```suggestion
```

File: docs/quickstart.md [hunk 8]
```diff
@@ -150,20 +150,20 @@ aea init --local
...
-svn checkout https://github.com/valory-xyz/open-aea/tags/v1.10.0/packages packages
+aea fetch PUBLIC_ID_OR_HASH
```
Comment (DavidMinarsch) [inline docs/quickstart.md:155]:
```suggestion
```

File: docs/quickstart.md [hunk 9]
```diff
@@ -150,20 +150,20 @@ aea init --local
...
 ```
```
Comment (DavidMinarsch) [inline docs/quickstart.md:156]:
```suggestion
```

File: docs/quickstart.md [hunk 10]
```diff
@@ -150,20 +150,20 @@ aea init --local
...
 
-This is a simple demo that introduces you to the main components of an AEA. 
+This is a simple demo that introduces you to the main components of an AEA.
 
 The fastest way to have your first AEA is to fetch one that already exists!
 
 ``` bash
-aea fetch open_aea/my_first_aea:0.1.0 --local
+aea fetch fetchai/my_first_aea:0.27.0:QmcmunuEAvaKfLCJsgeqF8M5dKARHixu4otUFVGbEenaCm --remote
 cd my_first_aea
```
Comment (DavidMinarsch) [inline docs/quickstart.md:167]:
```suggestion
aea fetch open_aea/my_first_aea:0.1.0:QmbyFaoHWikC8dHcpdTVboxVnkXX2vMgch4UUXmwCfJhRu --remote
cd my_first_aea
```

File: docs/quickstart.md [hunk 11]
```diff
@@ -321,7 +321,7 @@ The following test class replicates the preceding demo and tests it's correct be
 First, get the packages directory from the AEA repository (execute from the working directory which contains the <code>my_first_aea</code> folder):
 
 ``` bash
-svn export https://github.com/valory-xyz/open-aea.git/trunk/packages
+aea fetch fetchai/my_first_aea:0.27.0:QmcmunuEAvaKfLCJsgeqF8M5dKARHixu4otUFVGbEenaCm --remote
```
Comment (DavidMinarsch) [inline docs/quickstart.md:324]:
```suggestion
svn export https://github.com/valory-xyz/open-aea.git/trunk/packages
```
Replacing the single line and not changing the text won't work!

File: docs/decision-maker-transaction.md (hunk 3 above)
Comment (DavidMinarsch) [inline docs/decision-maker-transaction.md:3]:
```suggestion
First, import the libraries and the set the constant values. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
```

File: tests/test_docs/test_bash_yaml/md_files/bash-quickstart.md [hunk 12]
```diff
@@ -149,3 +149,12 @@ pipenv run pytest test.py
 ``` bash
 aea delete my_first_aea
 ```
+
+```bash
+aea fetch PUBLIC_ID_OR_HASH
+```
```
Comment (DavidMinarsch) [inline tests/test_docs/test_bash_yaml/md_files/bash-quickstart.md:155]:
```suggestion
```

File: docs/http-connection-and-skill.md [hunk 13]
```diff
@@ -26,7 +26,7 @@ cd my_aea
 Add the http server connection package:
 
 ``` bash
-aea add connection fetchai/http_server:0.22.0
+aea add connection fetchai/http_server:0.22.0:Qmdr2Ajx5BRBJCC6xP8wUNSCMbhStHHZaVgzkoSW6Pvu7S
```
Comment (DavidMinarsch) [inline docs/http-connection-and-skill.md:29]:
```suggestion
aea add connection fetchai/http_server:0.22.0:Qmdr2Ajx5BRBJCC6xP8wUNSCMbhStHHZaVgzkoSW6Pvu7S --remote
```

File: docs/ipfs_registry.md [hunk 14]
```diff
@@ -38,11 +38,11 @@ Packages can be downloaded using both extended public ids and hashes
 
 Add the signing protocol using
 
-`aea add protocol QmdrqznAYpnTWtLvLtLXiyMVr8PRfmj251puWbMZMNBw41`
+`aea add protocol QmWdCAiXpeT1tgrcf8uofKdQ6fLxMH77D5jwgQbbMo41df`
```
Comment (DavidMinarsch) [inline docs/ipfs_registry.md:41]:
```suggestion
`aea add protocol QmWdCAiXpeT1tgrcf8uofKdQ6fLxMH77D5jwgQbbMo41df --remote`
```

File: docs/ipfs_registry.md [hunk 15]
```diff
@@ -38,11 +38,11 @@ Packages can be downloaded using both extended public ids and hashes
...
 
 Or
 
-`aea add protocol open_aea/signing:1.0.0:QmdrqznAYpnTWtLvLtLXiyMVr8PRfmj251puWbMZMNBw41`
+`aea add protocol open_aea/signing:1.0.0:QmWdCAiXpeT1tgrcf8uofKdQ6fLxMH77D5jwgQbbMo41df`
```
Comment (DavidMinarsch) [inline docs/ipfs_registry.md:45]:
```suggestion
`aea add protocol open_aea/signing:1.0.0:QmWdCAiXpeT1tgrcf8uofKdQ6fLxMH77D5jwgQbbMo41df --remote`
```

File: docs/multiplexer-standalone.md (hunk 5 above)
Comment (DavidMinarsch) [inline docs/multiplexer-standalone.md:3]:
```suggestion
First, import the Python and application specific libraries and set the static variables. (Get the packages directory from the AEA repository `svn export https://github.com/valory-xyz/open-aea.git/trunk/packages`.)
```

File: docs/p2p-connection.md [hunk 16]
```diff
@@ -26,7 +26,7 @@ Create one AEA as follows:
 ``` bash
 aea create my_genesis_aea
 cd my_genesis_aea
-aea add connection valory/p2p_libp2p:0.1.0
+aea add connection valory/p2p_libp2p:0.1.0:QmYW7Q5iFN6RwWmHj7wvjUBZfxu3hJN4KvcTWW1AhdKhki
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:29]:
```suggestion
aea add connection valory/p2p_libp2p:0.1.0:QmYW7Q5iFN6RwWmHj7wvjUBZfxu3hJN4KvcTWW1AhdKhki --remote
```

File: docs/p2p-connection.md [hunk 17]
```diff
@@ -58,7 +58,7 @@ Create a second AEA:
 ``` bash
 aea create my_other_aea
 cd my_other_aea
-aea add connection valory/p2p_libp2p:0.1.0
+aea add connection valory/p2p_libp2p:0.1.0:QmYW7Q5iFN6RwWmHj7wvjUBZfxu3hJN4KvcTWW1AhdKhki
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:61]:
```suggestion
aea add connection valory/p2p_libp2p:0.1.0:QmYW7Q5iFN6RwWmHj7wvjUBZfxu3hJN4KvcTWW1AhdKhki --remote
```

File: docs/quickstart.md [hunk 18]
```diff
@@ -259,14 +255,14 @@ cd my_first_aea
 <br>
 Second, add the stub connection to the project.
 ``` bash
-aea add connection fetchai/stub:0.21.0
+aea add connection fetchai/stub:0.21.0:QmektTWmXcjThQd8md8nSYgLapR3Gks3n3WEzwAWQFgc4z
```
Comment (DavidMinarsch) [inline docs/quickstart.md:258]:
```suggestion
aea add connection fetchai/stub:0.21.0:QmektTWmXcjThQd8md8nSYgLapR3Gks3n3WEzwAWQFgc4z --remote
```

File: docs/quickstart.md [hunk 19]
```diff
@@ -259,14 +255,14 @@ cd my_first_aea
...
 ```
 <br>
 <b>Add the echo skill</b>
 <br>
 Third, add the echo skill to the project.
 ``` bash
-aea add skill fetchai/echo:0.19.0
+aea add skill fetchai/echo:0.19.0:QmReV2qgjE4mkY9n5iwJDoEcv174Sxbjhxo84YZpLyvcR2
```
Comment (DavidMinarsch) [inline docs/quickstart.md:265]:
```suggestion
aea add skill fetchai/echo:0.19.0:QmReV2qgjE4mkY9n5iwJDoEcv174Sxbjhxo84YZpLyvcR2 --remote
```

File: docs/registry.md [hunk 20]
```diff
@@ -22,14 +22,14 @@ Successfully added protocol 'open_aea/signing:1.0.0'.
 Once we have an agent, we can add individual components to the agent as so;
 
 ```
-aea add skill fetchai/echo:0.19.0
+aea add skill fetchai/echo:0.19.0:QmReV2qgjE4mkY9n5iwJDoEcv174Sxbjhxo84YZpLyvcR2
```
Comment (DavidMinarsch) [inline docs/registry.md:25]:
```suggestion
aea add skill fetchai/echo:0.19.0:QmReV2qgjE4mkY9n5iwJDoEcv174Sxbjhxo84YZpLyvcR2 --remote
```

File: tests/test_docs/test_bash_yaml/md_files/bash-gym-skill.md [hunk 21]
```diff
@@ -8,7 +8,7 @@ aea create my_gym_aea
 cd my_gym_aea
 ```
 ``` bash
-aea add skill fetchai/gym:0.20.0
+aea add skill fetchai/gym:0.20.0:QmdQisuqLhq81NHuas9ALYgP1tw1cZoE3vGueREgG8PgCZ
```
Comment (DavidMinarsch) [inline tests/test_docs/test_bash_yaml/md_files/bash-gym-skill.md:11]:
```suggestion
aea add skill fetchai/gym:0.20.0:QmdQisuqLhq81NHuas9ALYgP1tw1cZoE3vGueREgG8PgCZ --remote
```

File: tests/test_docs/test_bash_yaml/md_files/bash-http-connection-and-skill.md [hunk 22]
```diff
@@ -3,7 +3,7 @@ aea create my_aea
 cd my_aea
 ```
 ``` bash
-aea add connection fetchai/http_server:0.22.0
+aea add connection fetchai/http_server:0.22.0:Qmdr2Ajx5BRBJCC6xP8wUNSCMbhStHHZaVgzkoSW6Pvu7S
```
Comment (DavidMinarsch) [inline tests/test_docs/test_bash_yaml/md_files/bash-http-connection-and-skill.md:6]:
```suggestion
aea add connection fetchai/http_server:0.22.0:Qmdr2Ajx5BRBJCC6xP8wUNSCMbhStHHZaVgzkoSW6Pvu7S --remote
```

**Pull request merged**